udata/migrations/2020-06-11-add-resource-fs-filename.py

```diff
@@ -0,0 +1,108 @@
+'''
+The purpose here is to fill every resource with a fs_filename string field.
+The migration iterates on every file in the ressource storage.
+It extracts the dataset's slug from the file path, retrieves the dataset,
+and iterates on the dataset's resources to find the one owning the file.
+If one is found the fs_filename field is filled.
+If none is found the file is deleted.
+'''
+import logging
+from urllib.parse import urlparse
+
+from udata.core import storages
+from udata.models import Dataset, CommunityResource, Organization, User, Reuse
+from udata.core.dataset.models import get_resource
+
+log = logging.getLogger(__name__)
+
+
+def migrate(db):
+    log.info('Processing resources and community resources.')
+
+    resource_deletion_count = 0
+    avatar_deletion_count = 0
+    image_deletion_count = 0
+
+    # Parse every resources URL to make an index out of it
+    resource_index = dict()
+    datasets = Dataset.objects()
+    for dataset in datasets:
+        for resource in dataset.resources:
+            if resource.filetype == 'file':
```

Comment:
can you just do a quick check: see on prod if there's any resource with an URL in `static.data.gouv.fr` and `filetype != file`?